lib/linguist/languages.yml

```diff
@@ -4575,6 +4575,8 @@ Reason:
   extensions:
   - ".re"
   - ".rei"
+  - ".res"
```

Comment:
@lildude I'm having trouble checking there aren't any other languages using that extension on GitHub, mostly because of the large number of `.res` binary files. Is there a way to exclude binary files in GitHub searches?

Reply:
Unfortunately not, however the search results show quite a few files identified as other languages thanks to the other strategies. Many of those are XML files which means this PR is going to flip them all to Reason because the extension strategy has priority over the XML strategy.

The search result in the OP also returns quite a few files that don't look like Reason to me, though I know nothing about Reason.

Reply:
Indeed a few files that are not written in Reason, I don't know how I can filter them out in the search. Also, I think it's worth to say that this new syntax and the new file extensions for Reason/BuckleScript got available at the beginning of July 2020, so I'd say it's in early adoption phase.

Reply:
Thanks @brnrdog kicking this off, may I chime in here? 
This might sound confusing, but when we introduced the new Syntax for BuckleScript / Reason, we didn't give it a name yet, [until very recently](https://reasonml.org/blog/bucklescript-is-rebranding).

BuckleScript / Reason is rebranding to [ReScript](https://rescript-lang.org/), therefore the `.res` and `.resi` file extension should therefore be called `ReScript` as well.